src/analysis/special_functions/gamma.lean

```diff
@@ -1007,3 +1015,200 @@ end
+  rw [inv_neg, inv_one, neg_one_smul, ←interval_integral.integral_symm] at this,
+  convert this,
+  { ext1 x, rw mul_comm, congr;
+    { push_cast, ring } },
+  { ring }, { ring }
+end
+
+lemma beta_integral_eval_one_right {u : ℂ} (hu : 0 < re u) :
+  beta_integral u 1 = 1 / u :=
+begin
+  simp_rw [beta_integral, sub_self, cpow_zero, mul_one],
+  rw integral_cpow (or.inl _),
+  { rw [of_real_zero, of_real_one, one_cpow, zero_cpow,
+    sub_zero, sub_add_cancel],
+    rw sub_add_cancel,
+    contrapose! hu, rw [hu, zero_re] },
+  { rwa [sub_re, one_re, ←sub_pos, sub_neg_eq_add, sub_add_cancel] },
+end
+
+lemma beta_integral_scaled (s t : ℂ) {a : ℝ} (ha : 0 < a) :
+  ∫ x in 0..a, (x:ℂ) ^ (s - 1) * (a - x) ^ (t - 1) = a ^ (s + t - 1) * beta_integral s t :=
+begin
+  have ha' : (a:ℂ) ≠ 0, from of_real_ne_zero.mpr ha.ne',
+  rw beta_integral,
+  have A : (a:ℂ) ^ (s + t - 1) = a * (a ^ (s - 1) * a ^ (t - 1)),
+  { rw [(by abel : s + t - 1 = 1 + (s - 1) + (t - 1)),
+      cpow_add _ _ ha', cpow_add 1 _ ha', cpow_one, mul_assoc] },
+  rw [A, mul_assoc, ←interval_integral.integral_const_mul ((↑a) ^ _ * _),
+    ←real_smul, ←(zero_div a), ←div_self ha.ne',
+    ←interval_integral.integral_comp_div _ ha.ne', zero_div],
+  simp_rw interval_integral.integral_of_le ha.le,
+  refine set_integral_congr measurable_set_Ioc (λ x hx, _),
+  dsimp only,
+  rw mul_mul_mul_comm,
+  congr' 1,
+  { rw [←mul_cpow_of_real_nonneg ha.le (div_pos hx.1 ha).le, of_real_div, mul_div_cancel' _ ha'] },
+  { rw [(by push_cast : (1:ℂ) - ↑(x / a) = ↑(1 - x / a)),
+      ←mul_cpow_of_real_nonneg ha.le (sub_nonneg.mpr $ (div_le_one ha).mpr hx.2)],
+    push_cast,
+    rw [mul_sub, mul_one, mul_div_cancel' _ ha'] }
+end
+
+/-- Relation between Beta integral and Gamma function. Note that we haven't proved (yet) that
+the Gamma function has no zeroes, so we can't formulate this as a formula for the Beta function. -/
+lemma Gamma_mul_Gamma_eq_beta_integral {s t : ℂ} (hs : 0 < re s) (ht : 0 < re t) :
+  Gamma s * Gamma t = Gamma (s + t) * beta_integral s t :=
+begin
+  have conv_int := integral_pos_convolution (Gamma_integral_convergent hs)
+    (Gamma_integral_convergent ht)(continuous_linear_map.mul ℝ ℂ),
+  simp_rw continuous_linear_map.mul_apply' at conv_int,
+  have hst : 0 < re (s + t),
+  { rw add_re, exact add_pos hs ht },
+  rw [Gamma_eq_integral hs, Gamma_eq_integral ht, Gamma_eq_integral hst, Gamma_integral,
+    Gamma_integral, Gamma_integral, ←conv_int, ←integral_mul_right (beta_integral _ _)],
+  refine set_integral_congr measurable_set_Ioi (λ x hx, _),
+  dsimp only,
+  rw [mul_assoc, ←beta_integral_scaled s t hx, ←interval_integral.integral_const_mul],
+  congr' 1 with y:1,
+  push_cast,
+  suffices : cexp (-x) = cexp (-y) * cexp (-(x - y)),
+  { rw this, ring },
+  { rw ←complex.exp_add, congr' 1, abel },
+end
+
+/-- Recurrence formula for the Beta function. If we knew `Gamma (u + v + 1) ≠ 0` this would be an
```

Comment:
Can't you deduce non-vanishing of Gamma from the previous lemma? If `Gamma u = 0` for `u = s + it`, then taking `v = w -it` you get `Gamma (s+ i t + w -it) beta (u, w-it)` = 0 for all positive `w`. And you know that `Gamma (s + it + w - it)` is nonzero as we are along the positive real axis, so you get `beta (u, w-it) = 0` for all positive `w`.

But the function `z -> beta (u, z)` is holomorphic in the half-plane `Re z > 0`, so by the principle of isolated zeroes it is uniformly zero, a contradiction with `beta_integral_eval_one_right`. The nontrivial bit here is holomorphy of the integral (do we have API for that?). But maybe you have other plans to prove the nonvanishing of Gamma?

Reply:
I do indeed have other plans to get at this. Using the properties of the beta integral developed here + the Euler product formula for sine, I can show $\Gamma(s) \Gamma(1 - s) = \pi / \sin \pi s$ for all complex $s$, and then the nonvanishing is easy.

An alternative route is to show $\Gamma(s) \Gamma(1 - s) = \pi / \sin \pi s$ for *real* s first, then deduce it holds for complex s too by analytic continuation. I tried coding this up but it was a bit painful -- there are singularities at integers, and it is surprisingly awkward to show that $\mathbb{C} - \mathbb{Z}$ is connected! So I thought including this integration-by-parts argument was the lesser evil.

Reply:
Ok, fine. Can you move your comment `If we knew ...` to the beginning of the proof (this comment is not relevant for users of the lemma) and also mention there that this lemma will be used to prove nonvanishing of `Gamma` which justifies the not-so-nice argument.